This i386 code in libffi has been written now to assume a current darwin assembler
and .cfi directives. Bootstrapping with gcc-4.2 and the default "gas" assembler
no longer works. This patch re-enables the build of libffi by reverting the asm change
and disabling the unsupported features.

If important to someone, it might be possible to rebuild libffi with these features
re-enabled once a newer toolchain has been installed (clang/ld64/cctools).

https://trac.macports.org/ticket/61170

--- ./src/x86/sysv.S.orig
+++ ./src/x86/sysv.S
@@ -792,7 +792,7 @@
 
 #ifdef X86_DARWIN
 # define COMDAT(X)							\
-        .section __TEXT,__text,coalesced,pure_instructions;		\
+        .section __TEXT,__textcoal_nt,coalesced,pure_instructions;		\
         .weak_definition X;						\
         FFI_HIDDEN(X)
 #elif defined __ELF__ && !(defined(__sun__) && defined(__svr4__))
@@ -820,7 +820,7 @@
 #endif /* __PIC__ */
 
 /* Sadly, OSX cctools-as doesn't understand .cfi directives at all.  */
-
+#if 0
 #ifdef __APPLE__
 .section __TEXT,__eh_frame,coalesced,no_toc+strip_static_syms+live_support
 EHFrame0:
@@ -1120,7 +1120,7 @@
     .long    0
     .long    0
 #endif /* __APPLE__ */
-
+#endif /* if 0 */
 #endif /* ifndef _MSC_VER */
 #endif /* ifdef __i386__ */
 
